File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    check = [words[0]]
    answer = [0, 0]
    for i in range(1, len(words)):
        if words[i] not in check and check[-1][-1] == words[i][0]:
            check.append(words[i])
        else:
            answer = [i % n + 1,i // n + 1]
            break
    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        String now = words[0];
        HashSet<String> done = new HashSet<>();
        done.add(now);
        
        for(int i = 1; i < words.length; i++){
            if(now.charAt(now.length() - 1) != words[i].charAt(0) || done.contains(words[i])){
                return new int[]{i % n + 1, i / n + 1};
            }
            done.add(words[i]);  
            now = words[i];
        }

        return new int[]{0, 0};
    }
}